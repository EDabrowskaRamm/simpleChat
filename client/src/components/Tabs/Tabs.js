import React, { useState } from 'react'

import { TabsRoot, TabHeader, TabTitle, TabContent } from './Tabs.shards'

import Participants from '../Participants/Participants'

const getTabs = (participantsCount) => [
  {
    name: `Participants ${participantsCount}`,
    content: <Participants />
  },
  {
    name: 'Chat',
    content: 'asd2'
  }
]

const Tabs = ({ participantsCount = 0 }) => {
  const [activeIndex, setActiveIndex] = useState(0)
  const tabs = getTabs(participantsCount)

  return (
    <TabsRoot>
      <TabHeader>
        {tabs.map((tab, index) => <TabTitle
          key={index}
          isActive={index === activeIndex}
          onClick={() => setActiveIndex(index)}
          index={index}
        >
          {tab.name}
        </TabTitle>)}
      </TabHeader>
      <div>
        {tabs.map((tab, index) => <TabContent
          isActive={index === activeIndex}
          key={index}
        >
          {tab.content}
        </TabContent>)}
      </div>
    </TabsRoot>
  )
}

export default Tabs
