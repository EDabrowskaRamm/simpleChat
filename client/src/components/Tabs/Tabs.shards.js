import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const TabsRoot = styled.section`
  width: 100%;
`

export const TabHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: 120px;
  background-color: ${({ theme }) => theme.colors.background.secondary};
`

export const TabTitle = styled.p`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 50%;
  height: 100%;
  cursor: pointer;
  background-color: ${({ theme }) => theme.colors.background.secondary};
  border-bottom: ${({ theme }) => theme.commons.borders.main};

  ${({ isActive, theme }) => isActive && css`
    background-color: ${theme.colors.background.primary};
    border-top: ${theme.commons.borders.main};
    border-bottom: ${theme.commons.borders.transparent};
  `}

  ${({ isActive, index, theme }) => isActive && index === 0 && css`
    &:first-of-type {
      border-top-right-radius: 5px;
      border-top-left-radius: 0;
      border-right: ${theme.commons.borders.main};
    }
  `}

  ${({ isActive, index, theme }) => isActive && index === 1 && css`
    &:last-of-type {
      border-top-left-radius: 5px;
      border-top-right-radius: 0;
      border-left: ${theme.commons.borders.main};
    }
  `}

  &:hover {
    opacity: .9;
  }
`

export const TabContent = styled.div`
  display: none;

  ${({ isActive }) => isActive && css`
    display: block;
    width: 100%;
  `}
`
