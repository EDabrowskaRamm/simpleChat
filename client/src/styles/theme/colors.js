const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
