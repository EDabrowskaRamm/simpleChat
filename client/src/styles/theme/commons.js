import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
