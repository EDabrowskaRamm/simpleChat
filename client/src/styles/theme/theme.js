import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
