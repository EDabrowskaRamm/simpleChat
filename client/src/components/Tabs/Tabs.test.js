import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import Tabs from './Tabs'

import theme from '../../styles/theme/theme'
import initialState from '../../../__mocks__/fileMock'

expect.addSnapshotSerializer(serializer)

describe('renders Tabs', () => {
  it('Snapshot', () => {
    const mockStore = configureStore()

    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <ThemeProvider theme={theme}>
          <Tabs />
        </ThemeProvider>
      </Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
