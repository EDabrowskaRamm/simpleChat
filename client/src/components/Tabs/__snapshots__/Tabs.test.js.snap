// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tabs Snapshot 1`] = `
.emotion-21 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 120px;
  background-color: #ebebeb;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
  height: 100%;
  cursor: pointer;
  background-color: #ebebeb;
  border-bottom: 1px solid #e1e3e5;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  background-color: #FFF;
  border-top: 1px solid #e1e3e5;
  border-bottom: 1px solid transparent;
}

.emotion-0:first-of-type {
  border-top-right-radius: 9px;
  border-top-left-radius: 0;
  border-right: 1px solid #e1e3e5;
}

.emotion-0:hover {
  opacity: .9;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
  height: 100%;
  cursor: pointer;
  background-color: #ebebeb;
  border-bottom: 1px solid #e1e3e5;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.emotion-1:hover {
  opacity: .9;
}

.emotion-6 {
  display: none;
  display: block;
  width: 100%;
}

.emotion-5 {
  width: 100%;
  padding: 0 30px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 120px;
  padding-left: 12px;
  border-bottom: 1px solid #e1e3e5;
  text-transform: capitalize;
}

.emotion-20 {
  display: none;
}

.emotion-11 {
  padding-left: 17px;
  padding-bottom: 17px;
}

.emotion-11:first-of-type {
  padding-top: 25px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 11px 0 5px;
}

.emotion-7 {
  font-weight: bold;
  text-transform: capitalize;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  margin-right: 15px;
}

.emotion-8 {
  color: #979c9f;
}

.emotion-10 {
  -webkit-letter-spacing: 1.4px;
  -moz-letter-spacing: 1.4px;
  -ms-letter-spacing: 1.4px;
  letter-spacing: 1.4px;
  line-height: 1.6;
}

.emotion-18 {
  width: 100%;
  padding: 15px 0;
}

.emotion-17 {
  width: 100%;
  padding: 27px 33px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  border: 2px solid #868c97;
  border-radius: 9px;
}

.emotion-17:placeholder {
  color: #979c9f;
}

<section
  className="emotion-21"
>
  <div
    className="emotion-2"
  >
    <p
      className="emotion-0"
      disabled={false}
      onClick={[Function]}
    >
      Participants (2)
    </p>
    <p
      className="emotion-1"
      disabled={false}
      onClick={[Function]}
    >
      Chat
    </p>
  </div>
  <div>
    <div
      className="emotion-6"
    >
      <section
        className="emotion-5"
      >
        <div>
          <div
            className="emotion-3"
          >
            <p>
              Jane Downey
            </p>
          </div>
          <div
            className="emotion-3"
          >
            <p>
              michel sagan
            </p>
          </div>
        </div>
      </section>
    </div>
    <div
      className="emotion-20"
    >
      <section
        className="emotion-5"
      >
        <div>
          <div
            className="emotion-11"
          >
            <div
              className="emotion-9"
            >
              <p
                className="emotion-7"
              >
                Jane Downey
              </p>
               
              <span
                className="emotion-8"
              >
                12:58:24 
              </span>
            </div>
            <p
              className="emotion-10"
            >
              After losing his wife seven years earlier, the eccentric Dr. John Dolittle, famed doctor and veterinarian of Queen Victoriaâ€™s England, hermits himself away behind the high walls of Dolittle Manor with only his menagerie of exotic animals for company.
            </p>
          </div>
          <div
            className="emotion-11"
          >
            <div
              className="emotion-9"
            >
              <p
                className="emotion-7"
              >
                Jane Downey
              </p>
               
              <span
                className="emotion-8"
              >
                12:59:54 
              </span>
            </div>
            <p
              className="emotion-10"
            >
              But when the young queen falls gravely ill, a reluctant Dolittle is forced to set sail on an epic adventure to a mythical island in search of a cure, regaining his wit and courage as he crosses old adversaries and discovers wondrous creatures.
            </p>
          </div>
        </div>
        <form
          className="emotion-18"
          onSubmit={[Function]}
        >
          <input
            className="emotion-17"
            id="message"
            onChange={[Function]}
            placeholder="Message"
            type="text"
            value=""
          />
        </form>
      </section>
    </div>
  </div>
</section>
`;
